pubsub/azuresb: nil pointer dereference on receive

### Describe the bug

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x15bcdf2]

goroutine 126 [running]:
gocloud.dev/pubsub.(*Subscription).Receive(0xc000cec0e0, 0x1a73460, 0xc000ac8240, 0x0, 0x0, 0x0)
        /Users/massel/go/pkg/mod/gocloud.dev@v0.21.0/pubsub/pubsub.go:581 +0x552
```

### To Reproduce

Call Receive on azuresb subscription with no message.

## Expected behavior

Wait till there is a message

### Version

v0.21

### Additional context

azuresb's ReceiveBatch is as follow:

```
        rctx, cancel := context.WithTimeout(ctx, listenerTimeout)
	defer cancel()

	var message *driver.Message
	err := s.sbSub.ReceiveOne(rctx, servicebus.HandlerFunc(func(_ context.Context, sbmsg *servicebus.Message) error {
		metadata := map[string]string{}
		for key, value := range sbmsg.GetKeyValues() {
			if strVal, ok := value.(string); ok {
				metadata[key] = strVal
			}
		}
		// partitionID is only available if partitioning is enabled; otherwise,
		// it defaults to 0.
		var partitionID int16
		if sbmsg.SystemProperties != nil && sbmsg.SystemProperties.PartitionID != nil {
			partitionID = *sbmsg.SystemProperties.PartitionID
		}
		message = &driver.Message{
			Body:     sbmsg.Data,
			Metadata: metadata,
			AckID:    &partitionAckID{partitionID, sbmsg.LockToken},
			AsFunc:   messageAsFunc(sbmsg),
		}
		return nil
	}))

	// Mask rctx timeouts, they are expected if no messages are available.
	if err == rctx.Err() {
		err = nil
	}
	return []*driver.Message{message}, err
```

So basically  `ReceiveOne` will take at most `listenerTimeout` to return (1 second), if rctx is done, we return an a slice with a single element which is `nil`.

```
return []*driver.Message{message}, err
```

The returned slice is not empty, and the first element is nil (default value of `message`). 

This leads to a crash in the portable type `Receive` method here https://github.com/google/go-cloud/blob/master/pubsub/pubsub.go#L581

--- a/pubsub/azuresb/azuresb.go
+++ b/pubsub/azuresb/azuresb.go
@@ -452,7 +452,7 @@ func (s *subscription) ReceiveBatch(ctx context.Context, maxMessages int) ([]*dr
 		return nil, s.linkErr
 	}
 
-	// ReceiveOne will block until ctx is Done; we want to return after
+	// ReceiveOne will block until rctx is Done; we want to return after
 	// a reasonably short delay even if there are no messages. So, create a
 	// sub context for the RPC.
 	rctx, cancel := context.WithTimeout(ctx, listenerTimeout)
@@ -462,7 +462,7 @@ func (s *subscription) ReceiveBatch(ctx context.Context, maxMessages int) ([]*dr
 	// that aren't necessarily finished when Receive returns, which causes
 	// data races if Receive is called again quickly. ReceiveOne is more
 	// straightforward.
-	var message *driver.Message
+	var messages []*driver.Message
 	err := s.sbSub.ReceiveOne(rctx, servicebus.HandlerFunc(func(_ context.Context, sbmsg *servicebus.Message) error {
 		metadata := map[string]string{}
 		for key, value := range sbmsg.GetKeyValues() {
@@ -476,19 +476,19 @@ func (s *subscription) ReceiveBatch(ctx context.Context, maxMessages int) ([]*dr
 		if sbmsg.SystemProperties != nil && sbmsg.SystemProperties.PartitionID != nil {
 			partitionID = *sbmsg.SystemProperties.PartitionID
 		}
-		message = &driver.Message{
+		messages = append(messages, &driver.Message{
 			Body:     sbmsg.Data,
 			Metadata: metadata,
 			AckID:    &partitionAckID{partitionID, sbmsg.LockToken},
 			AsFunc:   messageAsFunc(sbmsg),
-		}
+		})
 		return nil
 	}))
 	// Mask rctx timeouts, they are expected if no messages are available.
 	if err == rctx.Err() {
 		err = nil
 	}
-	return []*driver.Message{message}, err
+	return messages, err
 }
 
 func messageAsFunc(sbmsg *servicebus.Message) func(interface{}) bool {

